lib/datadog/core/crashtracking/agent_base_url.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require_relative '../configuration/ext'
+
+module Datadog
+  module Core
+    module Crashtracking
+      module AgentBaseUrl
+        module_function
+
+        def resolve(agent_settings)
```

Comment:
Minor: I suggest not using `module_function` as it also makes this module something we can include but we have no plans of including it?

```suggestion
        def self.resolve(agent_settings)
```